map_frame: "base_footprint"
expected_update_rate: 20
map_resolution: 0.05
map_width: 10.0
map_height: 10.0

processors:
  - FootprintProcessor
  - PointCloud2Processor
  - GlobalPathProcessor
  - VelocityProcessor

PointCloud2Processor:
  base_frame: "base_footprint"
  input_data_type: "LaserScan"
  input_pcl_topic: "scan"
  output_pcl_topic: "transformed_pcl"
  map_height: 10.0
  map_width: 10.0
  map_resolution: 0.05
  update_rate: 15
  max_obstacle_height: 1.5
  data_value: 1  # 2^0

FootprintProcessor:
  footprint: 
            [[-0.105,-0.105],
             [-0.105, 0.105],
             [0.105,  0.105],
             [0.105, -0.105]]
  use_padding: true
  padding_size: 0.1
  base_frame: "base_footprint"
  map_height: 10.0
  map_width: 10.0
  map_resolution: 0.05
  udpate_rate: 15
  data_value: 8  # 2^3
  
VelocityProcessor:
  topic_name: "feedback_vel"
  expected_update_rate: 10
  map_height: 10.0
  map_width: 10.0
  map_resolution: 0.05
  udpate_rate: 10
  data_value: 16  ## 2^4

GlobalPathProcessor:
  base_frame: "base_footprint"
  path_topic: "global_planner/planner/plan"
  map_height: 10.0
  map_width: 10.0
  map_resolution: 0.05
  udpate_rate: 10
  data_value: 2  # 2^1
  goal_value: 32  # 2^5

# LocalPathProcessor:
#   base_frame: "base_footprint"
#   path_topic: "local_planner/planner/plan"
#   map_height: 10.0
#   map_width: 10.0
#   map_resolution: 0.05
#   udpate_rate: 10
#   data_value: 16  # 2^4

